### PR TITLE
Sonar Clean up in CoreHelp

### PR DESCRIPTION
**Pull request by Delphinus8821**
### Prerequisites

- [X] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first.

### Description of the Change
Clean up of code smells in the CoreHelp module.

### Why Should This Be In Core?
Removing bugs and code smells from CoreHelp.

### Benefits
Removing bugs and code smells from CoreHelp.

### Applicable Issues
#1022

### CODE REVIEW

File: CoreHelp/src/au/gov/asd/tac/constellation/help/preferences/HelpOptionsPanelController.java [hunk 1]
```diff
@@ -85,7 +85,7 @@ public boolean isValid() {
     public boolean isChanged() {
         final Preferences prefs = NbPreferences.forModule(HelpPreferenceKeys.class);
         final HelpOptionsPanel helpOptionsPanel = getPanel();
-        return !(helpOptionsPanel.getOnlineHelpOption() == prefs.getBoolean(HelpPreferenceKeys.HELP_KEY, HelpPreferenceKeys.ONLINE_HELP));
+        return (helpOptionsPanel.isOnlineHelpSelected() != prefs.getBoolean(HelpPreferenceKeys.HELP_KEY, HelpPreferenceKeys.ONLINE_HELP));
```
Comment (antares1470) [inline CoreHelp/src/au/gov/asd/tac/constellation/help/preferences/HelpOptionsPanelController.java:88]: The brackets can probably be omitted here

File: CoreHelp/src/au/gov/asd/tac/constellation/help/utilities/toc/TOCItem.java [hunk 2]
```diff
@@ -55,7 +55,7 @@ public String toString() {
      */
     @Override
     public boolean equals(final Object obj) {
-        return obj != null && obj instanceof TOCItem
+        return obj instanceof TOCItem
```
Comment (antares1470) [inline CoreHelp/src/au/gov/asd/tac/constellation/help/utilities/toc/TOCItem.java:58]: More just curious what rule was being addressed. Seems odd to remove a null check in this instance

File: CoreHelp/src/au/gov/asd/tac/constellation/help/utilities/toc/TreeNode.java [hunk 3]
```diff
@@ -196,9 +188,9 @@ public static TreeNode search(final TOCItem findItem, final TreeNode searchNode)
      */
     @Override
     public boolean equals(final Object obj) {
-        if (obj != null && obj instanceof TreeNode && ((TreeNode) (obj)).getData() != null && (((TreeNode) (obj)).getData().equals(data))) {
+        if (obj instanceof TreeNode && ((TreeNode) (obj)).getData() != null && (((TreeNode) (obj)).getData().equals(data))) {
             return true;
-        } else if (obj != null && obj instanceof TreeNode && ((TreeNode) (obj)).getData() == null && data == null) {
```
Comment (antares1470) [inline CoreHelp/src/au/gov/asd/tac/constellation/help/utilities/toc/TreeNode.java:201]: similar here

File: CoreHelp/src/au/gov/asd/tac/constellation/help/HelpServlet.java [hunk 4]
```diff
@@ -60,14 +60,14 @@
...
-    protected boolean getWasRedirect() {
-        return wasRedirect;
+    protected boolean isRedirect() {
+        return redirect;
     }
 
-    protected void setWasRedirect(final boolean wasRedirect) {
-        this.wasRedirect = wasRedirect;
+    protected static void setWasRedirect(final boolean wasRedirect) {
+        HelpServlet.redirect = wasRedirect;
```
Comment (antares1470) [inline CoreHelp/src/au/gov/asd/tac/constellation/help/HelpServlet.java:70]: bit confused by this one since the getter isn't static

**Review by antares1470 — COMMENTED** [review]
There were a few lingering sonar smells picked up. Not sure whether you intended to look at some of those?

File: CoreHelp/src/au/gov/asd/tac/constellation/help/utilities/toc/TOCItem.java (hunk 2 above)
Comment (Delphinus8821) [reply CoreHelp/src/au/gov/asd/tac/constellation/help/utilities/toc/TOCItem.java:58]: image
The explanation of the sonar rule is that null is not an instanceof anything so the null check is redundant.

File: CoreHelp/src/au/gov/asd/tac/constellation/help/HelpServlet.java (hunk 4 above)
Comment (Delphinus8821) [reply CoreHelp/src/au/gov/asd/tac/constellation/help/HelpServlet.java:70]: Not sure why it only picked up code smells on the setter. These are the code smells it wanted changed. 
image

File: CoreHelp/src/au/gov/asd/tac/constellation/help/utilities/toc/TOCItem.java (hunk 2 above)
Comment (antares1470) [reply CoreHelp/src/au/gov/asd/tac/constellation/help/utilities/toc/TOCItem.java:58]: Gotcha

File: CoreHelp/src/au/gov/asd/tac/constellation/help/HelpServlet.java (hunk 4 above)
Comment (antares1470) [reply CoreHelp/src/au/gov/asd/tac/constellation/help/HelpServlet.java:70]: Right so the variable is actually static hence the smells but yes it is rather odd that the getter wasn't picked up. Unsurprisingly, if you use the netbeans shortcuts to make the getter, it makes the method static so I wonder whether it simply hasn't realised for some reason

**Review by antares1470 — APPROVED** [review]

**Review by aldebaran30701 — APPROVED** [review]

**Pull request merged**